docs/DeveloperGuide.md

```diff
@@ -927,3 +927,70 @@ long inputs, whether it be for `n/[NAME]` or `d/[DESCRIPTION]` or `t/[TAG]` caus
+- For example, if the task list only contains a parent task that has an effort level of 50, with two subsections called "homework A" and "homework B" with 25 effort level each, when the user enters `schedule E/25 D/2023-05-04` and today is 2023-05-04, the displayed result 
+will only be the parent task and the subsection "homework A". If the user enters `schedule E/25 D/2023-05-05` then, the same parent task with subsection "homework B" will be displayed.
+- This method ensures that each subsection is treated as a somewhat individual task when scheduling, so that the main task's workload can be spread out. The scheduling algorithm will allocate the deadlines' subsections before the deadline date, even if that may result in 
+the total effort on a certain date exceeding daily effort level. Also, the events' subsections should occur between the events' start and end time. 
+- Moreover, allowing parent tasks to have deadlines and events as subsections will allow deadlines to be broken down into multiple deadlines and events to contain multiple events. Those components will be scheduled individually and displayed along with the parent tasks. 
+This ensures that the user have more flexibility in handling subsections.
+
+  
+### 7.4 Enabling `sort` by subsection
+
+- The application currently does not support sorting subsections so that they can be completed on different days.
+#### Possible Solution
+- One possible solution is to implement a `compareTo` function in the `Subtask` class. This is so that we can call `list.sort()` for the list of subsections in a task with a custom `compareTo` method.
+- The command will now look like `sort-sub INDEX`, where the index is the index of the main task in the task list currently displayed to the user. The result of the command will be that all the subsections
+of the selected parent task will be sorted. The sorting will sort subsections in the order of simple task > deadlines > events, followed by the effort level of the subsections, breaking ties using the subsection names' alphabetical order. This order is similar to the main tasks' sorting method.
+- The result of the sorted subtasks will be displayed with new indexes inside the parent task.
+- To sort all the tasks, use `sort-sub all/` and the command will call the sorting function on the subsection list of each of the main tasks displayed.
+
+
+### 7.5 Better error messages
+
+- Currently, there are still some error messages that are not specific enough. For example, for the edit command, if the user enters an index that is out of bounds, i.e. the index is negative or the index is larger than the maximum index currently displayed in the list, the error message will be:
+  ![text-wrap0](images/index-error.png)
+However, a more specific error would be `The index provided is invalid`. In the future, the error message will be changed to `The index provided is invalid` for index errors.
```

Comment:
Perhaps we can consider making an IndexOutOfBoundsException class for all these cases.

Reply:
ok sure! tbh i think when we execute the command, the command will have access to the task list so that it will compare the list with the index, and throw an error if there is index out of bounds. However, edit seems to catch and throw the correct error but included the more general error message. For the 0 index, it is because when the parser throws the error, the catcher throw a new exception but with message about the standard usage of command. Therefore, I feel that the problem is just how we chose the error message.

Reply:
Sure